src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1932,8 +1821,25 @@ private static TypeSymbolWithAnnotations InferResultNullability(Conversion conve
                     var methodOpt = conversion.Method;
                     if ((object)methodOpt != null && methodOpt.ParameterCount == 1)
                     {
+                        if (conversion.BestUserDefinedConversionAnalysis.Kind == UserDefinedConversionAnalysisKind.ApplicableInLiftedForm)
+                        {
+                            // PROTOTYPE(NullableReferenceTypes): Conversions: Lifted operator
+                            break;
+                        }
                         // PROTOTYPE(NullableReferenceTypes): Update method based on operandType.
-                        return methodOpt.ReturnType;
+                        var resultType = methodOpt.ReturnType;
+                        // Ensure converted type matches expected target type, specifically for lifted
+                        // conversions. See LocalRewriter.MakeConversionNode for similar handling
+                        // based on conversion.BestUserDefinedConversionAnalysis.ToType.
+                        if (!fromConversionNode)
+                        {
+                            if (resultType.TypeSymbol != targetType)
+                            {
+                                // PROTOTYPE(NullableReferenceTypes): Should handle nested nullability.
```

Comment:
>// PROTOTYPE(NullableReferenceTypes): Should handle nested nullability. [](start = 32, length = 71)

I think something should be done about top level nullability here as well, I suggest adjusting the comment. I think, for user-defined conversions, we should treat transition from resultType to target type as just another built-in conversion.

Reply:
Added comment.

---
In reply to: [169409574](https://github.com/dotnet/roslyn/pull/24917#discussion_r169409574) [](ancestors = 169409574)